app/src/main/java/edu/agh/susgame/front/ui/components/game/components/map/components/drawers/NodeDrawer.kt

```diff
@@ -28,30 +34,37 @@ fun NodeDrawer(
     pathBuilderState: PathBuilder,
     onInspectedNodeChange: (NodeId?) -> Unit,
 ) {
+    Box(
+        modifier = Modifier
+            .fillMaxSize()
+    ) {
+        val context = LocalContext.current
+        gameGraph.nodes.forEach { (key, node) ->
 
-    gameGraph.nodes.forEach { (key, node) ->
+            val imageResource = when (node) {
+                is Host -> R.drawable.host
+                is Router -> R.drawable.router
+                is Server -> R.drawable.server
+                else -> R.drawable.host
```

Comment:
To `else` jest niezbędne, bo przy typowym Javowym dziedziczeniu nie wiadomo ile implementacji posiada dana klasa abstrakcyjna/interfejs. 

W Kotlinie bardzo fajnie to jest rozwiązane `sealed class`. Jedyne implementacje takiej klasy mogą znajdować się w tym samym module.  Przykładowe rozwiązanie tego problemu (`Shift+Ctrl+A` -> `Apply Patch...`):

[Use_of_sealed_class.patch](https://github.com/user-attachments/files/17636692/Use_of_sealed_class.patch)